src/creatures/players/player.cpp

```diff
@@ -1715,13 +1715,7 @@ void Player::onCreatureAppear(std::shared_ptr<Creature> creature, bool isLogin)
 	Creature::onCreatureAppear(creature, isLogin);
 
 	if (isLogin && creature == getPlayer()) {
-		for (int32_t slot = CONST_SLOT_FIRST; slot <= CONST_SLOT_LAST; ++slot) {
-			std::shared_ptr<Item> item = inventory[slot];
-			if (item) {
-				item->startDecaying();
-				g_moveEvents().onPlayerEquip(getPlayer(), item, static_cast<Slots_t>(slot), false);
-			}
-		}
+		onEquipInventory();
```

Comment:
Please move this onEquipInventory to outside of this conditional (isLogin), this will fix the issue that the imbuements stopped to work when dying, needing to unequip and equip again to back to normal.

Reply:
Believe me the problem is not unequipping or not equipping, I will adjust